### PR TITLE
Add possibility to configure the concurrency for sharing and userlog

### PR DESCRIPTION
**Pull request by d7oc**
## Description
This PR adds the possibility to configure the concurrency for sharing and userlog services

## Related Issue
- https://github.com/owncloud/ocis/issues/10647

## Motivation and Context
Adjustabliity 

## How Has This Been Tested?
- test environment:
- test case 1:
- test case 2:
- ...

## Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [ ] Tests only (no source changes)

## Checklist:
- [x] Code changes
- [ ] Unit tests added
- [ ] Acceptance tests added
- [x] Documentation generated (`make docs`) and committed
- [ ] Documentation ticket raised: <link>
- [ ] Documentation PR created: <link>

### CODE REVIEW

**Review by kobergj — APPROVED** [review]

File: charts/ocis/docs/values.adoc.yaml [hunk 1]
```diff
@@ -1229,6 +1229,9 @@ services:
       sha: ""
       # -- Image pull policy
       pullPolicy:
+    # -- Maximum number of concurrent go-routines. Higher values can potentially get work done faster but will also cause 
+    # more load on the system. Values of 0 or below will be ignored and the default value will be used.
+    maxConcurrency: "1"
```
Comment (wkloucek) [inline charts/ocis/docs/values.adoc.yaml:1234]: frontend has this setting already in 

https://github.com/owncloud/ocis-charts/blob/fcbfec935673da46bc8e617366e2c97fd6b6e629/charts/ocis/values.yaml#L1187-L1188

File: charts/ocis/values.yaml [hunk 2]
```diff
@@ -1228,6 +1228,9 @@ services:
       sha: ""
       # -- Image pull policy
       pullPolicy:
+    # -- Maximum number of concurrent go-routines. Higher values can potentially get work done faster but will also cause 
+    # more load on the system. Values of 0 or below will be ignored and the default value will be used.
+    maxConcurrency: "1"
```
Comment (wkloucek) [inline charts/ocis/values.yaml:1233]: frontend has this setting already in 

https://github.com/owncloud/ocis-charts/blob/fcbfec935673da46bc8e617366e2c97fd6b6e629/charts/ocis/values.yaml#L1187-L1188

Comment (d7oc) [reply charts/ocis/values.yaml:1233]: Thanks. Overlooked that. Question is just if we should somehow "unify" the settings. In frontend the old setting is named `concurrency`. In the existing `storageusers` the setting is named `maxConcurrency`. I'm also not sure if that should stay that organised. There is only this specific setting below `events.consumer` and also `events` has only `consumer`. I'm not a big fan of having only one entry per level.

Comment (wkloucek) [reply charts/ocis/values.yaml]: For the frontend and events service it's really about how many event consumers should be spawned. That is probably different to the other concurrency settings.

**Review by wkloucek — APPROVED** [review]
The documentation is not really helpful to understand what's actually going on, but the same as the oCIS product, so it's fine.

**Comment by wkloucek** [comment]
@d7oc I also reflected https://github.com/owncloud/ocis/issues/10647#issuecomment-2498338980 in this PR.

Basically with oCIS 5, the sharing concurrency was `0`, meaning unlimited. So setting it to 20 should be a more sensible default than `1` and also was backed by @butonic 's loadtests on Kubernetes.

**Pull request merged**